From: zsien <quezhiyong@deepin.org>
Date: Fri, 15 Mar 2024 09:53:26 +0800
Subject: Fix can't switch input methods through the systray menu

Call focusIn only when input method is accepted,
so that focusIn will not be called when the
systray menu pops up.
---
 qt5/platforminputcontext/qfcitxplatforminputcontext.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/qt5/platforminputcontext/qfcitxplatforminputcontext.cpp b/qt5/platforminputcontext/qfcitxplatforminputcontext.cpp
index f89ca7f..93c0de8 100644
--- a/qt5/platforminputcontext/qfcitxplatforminputcontext.cpp
+++ b/qt5/platforminputcontext/qfcitxplatforminputcontext.cpp
@@ -523,7 +523,7 @@ void QFcitxPlatformInputContext::setFocusObject(QObject *object) {
         return;
     }
 
-    if (proxy) {
+    if (proxy && !shouldDisableInputMethod()) {
         proxy->focusIn();
         // We need to delegate this otherwise it may cause self-recursion in
         // certain application like libreoffice.
@@ -611,7 +611,7 @@ void QFcitxPlatformInputContext::createInputContextFinished(
     if (proxy->isValid() && !uuid.isEmpty()) {
         QWindow *window = focusWindowWrapper();
         setFocusGroupForX11(uuid);
-        if (window && window == w) {
+        if (window && window == w && !shouldDisableInputMethod()) {
             cursorRectChanged();
             proxy->focusIn();
         }
